src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/SpecValidationUtility.cs

```diff
@@ -276,9 +276,9 @@ private static void ValidateStandaloneSpec(PackageSpec spec, IEnumerable<string>
 
         private static void ValidateToolSpec(PackageSpec spec, IEnumerable<string> files)
         {
-            var packageDependencies = GetAllDependencies(spec).ToList();
+            IEnumerable<LibraryDependency> packageDependencies = GetAllDependencies(spec);
```

Comment:
The current implementation is more expensive because we'd end up calling `GetAllDependencies` multiple times which would allocate quite a bit.